### DISCUSSION TITLE
Why is `importers` mandatory?

### DISCUSSION THREAD

**Discussion by buzinas** [vercel/turborepo#6023] [Help] [answered]
I'm currently using pnpm with [`shared-workspace-lockfile=false`](https://pnpm.io/npmrc#shared-workspace-lockfile) and `pnpm` doesn't create an `importers` field on `pnpm-lock.yaml` and for some reason Turbo thinks it's mandatory:

```bash
 WARNING  Issues occurred when constructing package graph. Turbo will function, but some features may not be available:  1 error occurred:
        * error performing lockfile operation: Unable to convert to yaml: missing field `importers`
```

Checking the code, I can see that almost all fields are optional / have their serialization skipped when not present:

https://github.com/vercel/turbo/blob/2bb8c9a2477b5fb20b4711fe34412559bc85aab9/crates/turborepo-lockfiles/src/pnpm/data.rs#L27

What am I missing?

**Comment by anthonyshew** [comment, marked as answer]
If you're still seeing this on latest, could you open an issue? 🙏

**Comment by LinnJS** [comment]
Linking the posted issue I found for visibility.

https://github.com/vercel/turbo/issues/6024
